### ISSUE TITLE
Repeater Fields again

### ISSUE THREAD

**Issue by AndreasFaust** [dadish/ProcessGraphQL#19]
Thank you for this amazing module! In my view, it's exactly what ProcessWire needs to stay relevant. I use it to combine its great infrastructure with the React-framework "Gatsby JS". It's a pleasure to build on PW without writing a single line of PHP. The only thing, that is missing, is support for Repeater fields, on which the also important "Fieldset Page" and "Repeater Matrix" depend. 
Do you plan to add it in the future? Is it even possible to do so?

**Comment by dadish** [comment]
There is a way to add support for almost any field via [third-party modules](https://github.com/dadish/ProcessGraphQL#third-party-fieldtypes-support). I'm sure Repeater could be implemented with it. However the focus right now is to migrate away from [Youshido/graphql](https://github.com/youshido-php/GraphQL#looking-for-maintainers), because, it seems like is not maintained anymore.

So the support for Repeaters is not gonna land from me in the near future.

**Comment by klaussa** [comment]
@dadish What would it take to continue this module ?

Kind Regards
Klauss

**Comment by dadish** [comment]
Hey @klaussa. Right now, with very limited time I have, I'm trying to migrate the current implementation from [youshido/graphql](https://github.com/youshido/graphql) to [webonyx/graphql-php](https://github.com/webonyx/graphql-php), because it seems like the former is not maintained anymore.

It will most likely require the rewrite of most of the code, so it'll take some time. Can't promise you any timelines on when it's gonna be done, sorry.

**Comment by MrSnoozles** [comment]
This module is incredible, dadish. My utmost respect to you for writing it.
Did you start already on the migration to webonyx or is that something you would like someone else to jump in to? I looked into it this weekend, but I couldn't get more than the most trivial examples working yet.

**Comment by dadish** [comment]
Thanks @MrSnoozles! Unfortunately I haven't commit any code yet, regarding the migration to webonyx/graphql-php. I have started to learn the webonyx library and play with it a little. Seems like I'm grasping their API. Looks like it's doable, just needs a lot of code rewrite.

You're absolutely free to try the migration yourself. The repository has tests that might give you some confidence. As long as tests pass, you're on the right track. You can check the [development](https://github.com/dadish/ProcessGraphQL/wiki/Development) wiki for guidelines. Feel free to ask any questions if you have.

**Comment by robots4life** [comment]
Was at a Vue workshop recently, all the people there raving about JS in the Backend and Frontend, but over lunch a pretty relaxed man mentioned "it is not only about JS everywhere, things are not really ready despite all the hype, for now I get great results with PHP and JS, simple and super fast, I do this with ProcessWire, heard of it?". That was 2 days ago.

Had a look and just reading through the docs I know I already totally love ProcessWire for all the default options and possibilities that come with it without all the issues or need for extras. Now since I am learning JS and Vue naturally I wondered if ProcessWire has a way to expose the content as JSON data. Here I am.

Bottom line, you do this in your free time.
Much respect for that.
Can I send you some $ to help you with your efforts to migrate to webonyx/graphql-php?
Are you sure that webonyx/graphql-php will be maintained in the future to avoid a similar scenario?

**Everyone**, let's support this man with $, even if it is just a bit, sure a "thanks" here and a "great job" there is nice, though sending a bit of $ can make a change if nothing at all.
Feel free to email me if you like.
Email address is on profile page.

**Comment by dadish** [comment]
Hey @robots4life! I'm always happy to hear that people are interested in this module. The money is not a problem at all, though thanks for the offer. I have a full time job at the moment that eats up all my cognitive resources each day 😄.

But that never stopped me before, all I needed to hear is that people are interested in it. I will get it done eventually. You can safely use the module as it is right now. There shouldn't be any breaking changes after I migrate to the new graphql library. It should continue to work as before. Just make sure you don't use the youshido-php/graphql directly.

Speaking of webonyx/graphql-php. There is no way I can be sure that it will be maintained in the future. But I'm confident in it because it's very popular and actively maintained. Also lots of other libraries, like popular laravel-graphql plugins use webonyx/graphql-php. So we will probably be safe with this one.

**Comment by robots4life** [comment]
Ok, glad to hear!
Yes, very much so interested!
Was quite happy to find out about it.
The good thing is that is gives devs all the freedom to use whatever they like in the frontend.
Perhaps ProcessWire will add this functionality to core in the future.
Given all the hype in various frontend frameworks and generally letting the dev decide what to do with the data is a great thing.

About to sign up to ProcessWire forums and start learning how to use it.
Once I know my way round that I hope to use your module and see what can be done with it in the frontend.
Saw your https://github.com/dadish/skyscrapers-app demo and am amazed.
Seems like ProcessWire can really easily be used for a lot of different scenarios without too much hassle.

Also great to hear that I can start using your module now without having to change things once you migrate to webonyx/graphql-php, nice one!

Yes, https://github.com/webonyx/graphql-php seems to be maintained and has lots of commits and active users. Fingers crossed it will stay like that.

All the best and thank you for your swift reply.
Keep rocking!

**Comment by dadish** [comment]
Just made a release of the [latest version](https://github.com/dadish/ProcessGraphQL/releases/tag/v1.0.2) of this module. We now use `webonyx/graphql-php` and support FieldtypeRepeater out of the box.

Go check it out and come back with your feedbacks! I'll close this one because Repeaters are supported now. But if you will encounter bugs regarding repeaters please open another issue about a particular bug in repeaters.
